dex-operator/pkg/stub/handler.go

```diff
@@ -0,0 +1,202 @@
+package stub
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/operator-sdk-samples/dex-operator/pkg/apis/auth/v1alpha1"
+
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func NewHandler() sdk.Handler {
+	return &Handler{}
+}
+
+type Handler struct {
+	// Fill me
+}
+
+func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
+	switch o := event.Object.(type) {
+	case *v1alpha1.Dex:
+		if event.Deleted {
+			return nil
+		}
+		dep := newDexPod(o)
+		err := sdk.Create(dep)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			logrus.Errorf("Failed to create dex pod : %v", err)
+			return err
+		}
+		err = sdk.Create(newDexPod(o))
+		if err != nil && !errors.IsAlreadyExists(err) {
+			logrus.Errorf("Failed to create dex pod : %v", err)
+			return err
+		}
+		err = sdk.Create(newDexService(o))
+		if err != nil && !errors.IsAlreadyExists(err) {
+			logrus.Errorf("Failed to create dex pod : %v", err)
+			return err
+		}
+		err = sdk.Get(dep)
+		if err != nil {
+			return fmt.Errorf("failed to get deployment: %v", err)
+		}
+		size := o.Spec.Size
+		if *dep.Spec.Replicas != size {
+			dep.Spec.Replicas = &size
+			err = sdk.Update(dep)
+			if err != nil {
+				return fmt.Errorf("failed to update deployment: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+func newDexService(cr *v1alpha1.Dex) *corev1.Service {
+	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{"app": "dex"},
+			Ports: []corev1.ServicePort{corev1.ServicePort{
+				Name:       "dex",
+				Protocol:   corev1.ProtocolTCP,
+				Port:       5556,
+				TargetPort: intstr.FromInt(5556),
+				NodePort:   32000,
+			}},
+		},
+	}
+}
+
+func newDexConfigMap(cr *v1alpha1.Dex) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		},
+		Data: map[string]string{
+			"config.yaml": `|
```

Comment:
Should be read from a file in a pre-defined location, or file location should be passed in as a parameter.